pkg/sql/bulkmerge/merge_processor.go

```diff
@@ -156,8 +165,117 @@ func (m *bulkMergeProcessor) handleRow(row rowenc.EncDatumRow) (rowenc.EncDatumR
 
 // Start implements execinfra.RowSource.
 func (m *bulkMergeProcessor) Start(ctx context.Context) {
-	m.StartInternal(ctx, "bulkMergeProcessor")
-	m.input.Start(ctx)
+	ctx = m.StartInternal(ctx, "bulkMergeProcessor")
+	// TODO(annie): also find a way to process input/get task from coordinator
+	if err := m.mergeSSTs(ctx, 0); err != nil {
+		// TODO(annie): need to do this?
+		m.MoveToDraining(err)
+	}
+	m.MoveToDraining(nil)
+}
+
+func getKeysForTask(task int, splits []roachpb.Key) (roachpb.Key, roachpb.Key) {
+	if task == 0 {
+		return nil, splits[0]
+	}
+	last := len(splits) - 1
+	if task == last {
+		return splits[task], nil
+	}
+	if task > last {
+		panic(errors.AssertionFailedf("bad"))
+	}
+	return splits[task-1], splits[task]
+}
+
+// This is stolen from roachpb/data.go where we check if spans overlap.
+func isOverlapping(startKey, endKey roachpb.Key, sst execinfrapb.BulkMergeSpec_SST) bool {
+	otherStartKey, otherEndKey := sst.StartKey, sst.EndKey
+	if len(endKey) == 0 && len(otherEndKey) == 0 {
+		return startKey.Equal(otherStartKey)
+	} else if len(endKey) == 0 {
+		return bytes.Compare(startKey, otherStartKey) >= 0 && bytes.Compare(startKey, otherEndKey) < 0
+	} else if len(otherEndKey) == 0 {
+		return bytes.Compare(otherStartKey, startKey) >= 0 && bytes.Compare(otherStartKey, endKey) < 0
+	}
+	return bytes.Compare(endKey, otherStartKey) > 0 && bytes.Compare(startKey, otherEndKey) < 0
+}
+
+func (m *bulkMergeProcessor) mergeSSTs(ctx context.Context, taskID int) error {
+	// TODO(annie): get task from coordinator (look into how flows w multiple processors do it)
+	startKey, endKey := getKeysForTask(taskID, m.spec.Splits)
+	var storeFiles []storageccl.StoreFile
+	outputName := fmt.Sprintf("%s%d.sst", *m.spec.OutputUri, taskID)
+	dest, err := m.flowCtx.Cfg.ExternalStorageFromURI(ctx, outputName, username.RootUserName())
+	if err != nil {
+		return err
+	}
+	// Find the ssts that overlap with the given task's key range.
+	for _, sst := range m.spec.Ssts {
+		var store cloud.ExternalStorage
+		uri, err := url.Parse(sst.Uri)
+		if err != nil {
+			return err
+		}
+		// TODO(annie): cache
+		store, err = m.flowCtx.Cfg.ExternalStorageFromURI(ctx, sst.Uri, username.RootUserName())
```

Comment:
When opening the store, I think the rest of the logic is assuming it was opened with the path stripped off the basename.

Right now if sst.Uri was 'nodelocal://3/import/1333.sst', the store would try to open 'nodelocal://3/import/1333.sst/import/1333.sst' since it concatenates paths to the path used when opening the external storage.

Reply:
ah right this was the uri construction bit you mentioned earlier; done